xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/servletEngine/ServletContainerExecutor.java

```diff
@@ -170,6 +173,77 @@ private void startContainer() throws Exception
         start(this.servletContainer, this.testConfiguration);
     }
 
+    private void throwExceptionForServletEngineNotYetSupported(TestConfiguration testConfiguration)
+    {
+        throw new RuntimeException(String.format("Servlet engine [%s] is not yet supported!",
+            testConfiguration.getServletEngine()));
+    }
+
+    private GenericContainer getServletContainer(TestConfiguration testConfiguration)
+    {
+        final String baseImageName;
+
+        final GenericContainer container;
+
+        switch (testConfiguration.getServletEngine()) {
+            case TOMCAT:
+                baseImageName = String.format("tomcat:%s", testConfiguration.getServletEngineTag() != null
+                    ? testConfiguration.getServletEngineTag() : LATEST);
+                break;
+
+            case JETTY:
+                baseImageName = String.format("jetty:%s", testConfiguration.getServletEngineTag() != null
+                    ? testConfiguration.getServletEngineTag() : LATEST);
+                break;
+
+            case WILDFLY:
+                baseImageName = String.format("jboss/wildfly:%s", testConfiguration.getServletEngineTag() != null
+                    ? testConfiguration.getServletEngineTag() : LATEST);
+                break;
+
+            default:
+                this.throwExceptionForServletEngineNotYetSupported(testConfiguration);
+                baseImageName = null;
+        }
+
+        // we want to use libreoffice so we need a custom image
+        if (testConfiguration.isOffice()) {
+            // name of the image that we will create
```

Comment:
This comment is not very useful... What would be useful is to explain why you're using this algorithm to compute the servlet engine image name. Also it would be easier to read if you use String.format() instead of Arrays.asList(). That would also remove the need to use one CUSTOM_IMAGE_NAME_SEPARATOR and do not needed replacement.

`String.format("xwiki_%s_office", baseImageName.replace(':', CUSTOM_IMAGE_NAME_SEPARATOR)`

Now I'm not sure this is going to yield nice image names. I think I would simply prefer to use:

`String.format("xwiki-%s-office", <name of ServletEngine enum>)`

e.g. `xwiki-tomcat-office`, etc.

Reply:
I'm ok with your last proposal, but I'd personally keep the image tag in the name. It might give us some headache if we'd expect to use another image by trying another tag and it yield us to use the same image after all because of that..

Reply:
Well I've never seen this done anywhere. Look at all Dockerfiles that use a FROM. They never put the original tag in the new image name. At least I've never seen it done like this. For me it's the new tag that should indicate this.

Reply:
> For me it's the new tag that should indicate this.

You're completely right. I'll use the information to put a tag to the image then.